### PR TITLE
Batching/fetch pending identities at startup

### PR DESCRIPTION
**Pull request by Dzejkop**
## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -110,6 +110,15 @@ impl App {
 
         let database = Arc::new(database);
 
+        // Await for all pending transactions
+        let pending_identities = identity_manager.fetch_pending_identities().await?;
+
+        for pending_identity_tx in pending_identities {
+            // Ignores the result of each transaction - we only care about a clean slate in
+            // terms of pending transactions
+            drop(identity_manager.mine_identities(pending_identity_tx).await);
```
Comment (Dzejkop) [inline src/app.rs:119]: JSYK I'm more partial to `let _ = ...` but clippy::pedantic complains about it

File: src/ethereum/write_dev/mod.rs [hunk 2]
```diff
@@ -284,10 +288,14 @@ impl Provider {
     }
 
     #[instrument(level = "debug", skip_all)]
-    async fn mine_transaction(&self, tx: TransactionId) -> Result<(), TxError> {
-        let tx_hash = hex::decode(&tx.0).map_err(|err| TxError::Parse(Box::new(err)))?;
+    async fn fetch_pending_transactions(&self) -> Result<Vec<TransactionId>, TxError> {
+        // TODO: This implementation requires changes in the smart contract
+        Ok(vec![])
```
Comment (iamrecursion) [inline src/ethereum/write_dev/mod.rs:293]: Can you detail what change needs to be made here?

**Review by iamrecursion — APPROVED** [review]
LGTM, but I'd like more detail on what change is needed on the contract side.

File: src/ethereum/write_dev/mod.rs (hunk 2 above)
Comment (Dzejkop) [reply src/ethereum/write_dev/mod.rs:293]: Sure

**Pull request merged**